wfc3tools/embedsub.py

```diff
@@ -91,28 +91,37 @@ def embedsub(files):
         flt[1].header['sizaxis1'] = yaxis
         flt[1].header['sizaxis2'] = xaxis
 
-        for i in range(1, 4):
-            flt[i].header['crpix1'] = crpix1 + x1 - 1
-            flt[i].header['crpix2'] = crpix2 + y1 - 1
-            flt[i].header['ltv1'] = 0.0
-            flt[i].header['ltv2'] = 0.0
+        # loop through each possible fits extension and modify keywords if they exist
+        for i in range(1, len(flt)):
+            if 'CRPIX1' in flt[i].header:
+                flt[i].header['crpix1'] = crpix1 + x1 - 1
+                flt[i].header['crpix2'] = crpix2 + y1 - 1
+            if 'LTV1' in flt[i].header:
+                flt[i].header['ltv1'] = 0.0
+                flt[i].header['ltv2'] = 0.0
 
         # set the header value of SUBARRAY to False since it's now
         # regular size image
         flt[0].header['SUBARRAY'] = False
-
+        
         # Now write out the SCI, ERR, DQ extensions to the full-chip file
-        hdulist = fits.HDUList()
-        hdulist.append(fits.ImageHDU(flt[0].data, header=flt[0].header))
-        hdulist.append(fits.ImageHDU(sci, header=flt[1].header, name='SCI'))
-        hdulist.append(fits.ImageHDU(err, header=flt[2].header, name='ERR'))
-        hdulist.append(fits.ImageHDU(dq, header=flt[3].header, name='DQ'))
-
-        if not uvis:
-            hdulist.append(fits.ImageHDU(dq, header=flt[4].header, name='SAMP'))
-            hdulist.append(fits.ImageHDU(dq, header=flt[5].header, name='TIME'))
-
-        hdulist.writeto(full, clobber=False)
+        hdu_list = fits.open(filename)
```

Comment:
i think since 'flt' represents the input file opened in read-only mode already, this line can be omitted. flt[i].header will be the correct header already (no need to do hdu_list[i].header = flt[i].header), and the arrays can be assigned directly to flt[1], flt[2] etc, and then it can be written out. i'm going to change this